spacepy/__init__.py

```diff
@@ -284,31 +284,22 @@ def _write_defaults(rcfile, defaults, section='spacepy'):
             else:
                 nextsec = None
             #Read only this section for lines matching default values
-            present = []
             for l in rclines[thissec:nextsec]:
                 #For each key, does the line match, commented or not?
                 for k in writeme:
                     if re.match(r'(#\s?)?{key}\s?[:=]'.format(key=k),
                                 l):
                         writeme.remove(k)
                         break
-        f.seek(startpos) #Back to start of file
-        #Scan to just after section header. Text mode, can only seek() to tell()
-        if thissec != -1:
-            while f.readline() != '[{section}]\n'.format(section=section):
-                pass
-        if sys.platform == 'win32':
-            #Tickle the file for read/write switch
-            #https://stackoverflow.com/questions/11176724/python-file-operations
-            f.seek(0, 2)
-        #Write default values for anything not read
-        for k in sorted(writeme):
-            f.write(("#SpacePy {ver} default {key}: {value}\n"
-                     "#{key}: {value}\n").format(
-                         key=k, value=defaults[k], ver=__version__))
-        #And write all the remaining lines from the section header to end
-        if writeme:
-            f.writelines(rclines[thissec+1:])
+        #This is what we'll actually write to the file
+        writeme_verb = (f"#SpacePy {__version__} default {k}: {defaults[k]}\n"
+                        f"#{k}: {defaults[k]}\n"
+                        for k in sorted(writeme))
```

Comment:
This had me confused for a bit; I think it would be clearer as a list comprehension instead of a generator expression, since ultimately it's added into a list.